pages/index.js

```diff
@@ -13,10 +13,19 @@ export default function Home() {
     setItems(items.filter((item) => item.id !== id));
   }
 
+  function handleEditItem(editItem) {
+    const editedItem = items.filter((item) => item.id !== editItem.id);
+    setItems([editItem, ...editedItem]);
```

Comment:
I am not sure if this works, as the `const editedItem` actually descripes all the "Old" items which are not being edited. 
In the Setter function you should be adding the new item to the current "items" array in your state.
What you actually want to do is:
Check if there is an "old version of your `updatedItem` in your array 
- if yes, you want to replace it with the updatedItem

So what needs to go into this function?
1. create an array of your items that are updated. (e.g. `const updatedItemList`)
2. inside of this const you should map over your state-array(`items`), the items and inside of the map function body you ask
3. IF the id of your item (in your array `items`) === the id of your `updatedItem`, then you want to return the `updatedItem`(return updatedItem;).
4. if its not equal you want to return the item as it was (return item;)

After you mapped you still want to update your state in this function, where you only need to pass the updatedItemList, as you have already done all the work before :)

Reply:
Hey Merle, the function works like that. I changed the name of the const editedItem to make it more clear which items are used